chain/store/store.go

```diff
@@ -737,11 +775,15 @@ func (cs *ChainStore) GetPruningStartHeight() (uint32, uint32) {
 
 func (cs *ChainStore) getPruningStartHeight() (uint32, uint32) {
 	var pruningStartHeight, refCountStartHeight uint32
+	syncRootHeight := cs.getSyncRootHeight()
 
 	heightBuffer, err := cs.st.Get(db.TrieRefCountHeightKey())
 	if err != nil {
 		log.Info("get height of trie counted error:", err)
 		refCountStartHeight = 0
+		if syncRootHeight != 0 {
+			refCountStartHeight = syncRootHeight + 1
```

Comment:
It's better not modify logic here because it couples fast pruning logic with pruning logic. A better way is to set `TrieRefCountHeightKey`, `TrieCompactHeightKey`, etc after fast sync succeeds